src/Collection/Collection.service.ts

```diff
@@ -1,16 +1,129 @@
+import { HTTPError, STATUS_CODES } from '../common/HTTPError'
 import { collectionAPI } from '../ethereum/api/collection'
 import { isPublished } from '../utils/eth'
+import { isManager as isTPWManger } from '../ethereum/api/tpw'
+import { FactoryCollection } from '../ethereum/FactoryCollection'
+import { Ownable } from '../Ownable'
+import { CollectionAttributes, FullCollection } from './Collection.types'
+import { toDBCollection, decodeTPCollectionURN } from './utils'
+import { Collection } from './Collection.model'
 
+const DAY = 1000 * 60 * 60 * 24
 export class CollectionService {
   isLockActive(lock: Date | null) {
     if (!lock) {
       return false
     }
 
     const deadline = new Date(lock)
-    deadline.setDate(deadline.getDate() + 1)
+    return deadline.getTime() + DAY > Date.now()
+  }
+
+  private async checkIfNameIsValid(
+    id: string,
+    name: string,
+    urn_suffix?: string
+  ): Promise<void> {
+    if (!(await Collection.isValidName(id, name.trim(), urn_suffix))) {
+      throw new HTTPError(
```

Comment:
To discuss: For both this method and the ones below. Throwing `HTTPError`s seems a bit of a implementation detail right?
I don't want to make the code over-complex without needing to, but we just coupled some of the `CollectionService` to the Router, as it wouldn't make sense for this error to be thrown anywhere else.

So for example `checkIfNameIsValid` looks like a router method, as is just using `Collection.isValidName` and throwing an `HTTPError`

I understand why this is here, and the obvious solution would be to throw a more generalized error and then catching on the router and re-throwing an HTTPError (which might be hard for STATUS_CODES)

So I don't mean to change the implementation but wanted to know what you though of this and if you had it in mind when writing the code